parsl/tests/test_scaling/test_missing_heartbeat_3262.py

```diff
@@ -0,0 +1,91 @@
+import os
+import signal
+import time
+
+import pytest
+import zmq
+
+import parsl
+from parsl.channels import LocalChannel
+from parsl.config import Config
+from parsl.executors import HighThroughputExecutor
+from parsl.launchers import SimpleLauncher
+from parsl.providers import LocalProvider
+
+T_s = 1
+
+
+def fresh_config():
+    htex = HighThroughputExecutor(
+               heartbeat_period=1 * T_s,
+               heartbeat_threshold=3 * T_s,
+               label="htex_local",
+               worker_debug=True,
+               cores_per_worker=1,
+               encrypted=False,
+               provider=LocalProvider(
+                   channel=LocalChannel(),
+                   init_blocks=0,
+                   min_blocks=0,
+                   max_blocks=0,
+                   launcher=SimpleLauncher(),
+               ),
+           )
+    c = Config(
+        executors=[htex],
+        strategy='none',
+        strategy_period=0.5,
+    )
+    return c, htex
+
+
+@parsl.python_app
+def app():
+    return 7
+
+
+@pytest.mark.local
+@pytest.mark.parametrize("msg",
+                         (b'FuzzyByte\rSTREAM',  # not JSON
+                          b'{}',  # missing fields
+                          b'{"type":"heartbeat"}',  # heartbeat without ID
+                          )
+                         )
+def test_bad_messages(try_assert, msg):
+    """This tests that a heartbeat arriving after a manager
+    has expired due to missing heartbeats (due to transient
+    delay) does not kill the interchange - issue #3262 and
+    issue #3632.
+    """
+
+    c, htex = fresh_config()
+
+    with parsl.load(c):
+
+        # send a bad message into the interchange on the
+        # task_outgoing worker channel, and then check that
+        # we can scale out a block and run a task.
+
+        (task_port, result_port) = htex.command_client.run("WORKER_PORTS")
+
+        context = zmq.Context()
+        channel_timeout = 10000  # in milliseconds
+        task_channel = context.socket(zmq.DEALER)
+        task_channel.setsockopt(zmq.LINGER, 0)
+        task_channel.setsockopt(zmq.IDENTITY, b'testid')
+
+        task_channel.set_hwm(0)
+        task_channel.setsockopt(zmq.SNDTIMEO, channel_timeout)
+        task_channel.connect(f"tcp://localhost:{task_port}")
+
+        task_channel.send(msg)
```

Comment:
it's a repeat of code in the process worker pool to initialize this same "tasks flowing from interchange to worker pool" zmq socket.

If refactoring, I think it would make sense to put that into its own module that is then usable by both the real process worker pool and tests that are mocking parts of the worker pool (what this test does) - along the lines of the zmq_pipes module at https://github.com/Parsl/parsl/blob/master/parsl/executors/high_throughput/zmq_pipes.py for the submit side.